Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs

```diff
@@ -46,61 +51,88 @@ public Active2dViewVisibilityContext(Document hostDocument, Document targetDocum
         /****              Public methods               ****/
         /***************************************************/
 
+        [Description("Method used to return a RenderNodeAction.")]
+        [Input("node", "ElementNode to be processed.")]
+        [Output("action", "RenderNodeAction.")]
         public RenderNodeAction OnElementBegin2D(ElementNode node)
         {
             return base.OnElementBegin(node.ElementId);
         }
 
         /***************************************************/
 
+        [Description("Method used to return a RenderNodeAction.")]
+        [Input("node", "ElementNode to be processed.")]
         public void OnElementEnd2D(ElementNode node)
         {
+            throw new NotImplementedException();
```

Comment:
This method is empty on purpose: it says 'do nothing if you get here', which is very different from 'throw an exception if you get here' 😉 so please bring it back to the original state - refers to all cases where you have added the exception.